addon/v1alpha1/types_clustermanagementaddon.go

```diff
@@ -175,10 +185,146 @@ type PlacementStrategy struct {
 	// User can override the configuration by updating the managedClusterAddon directly.
 	// +optional
 	Configs []AddOnConfig `json:"configs,omitempty"`
+	// The rollout strategy to apply addon configurations change.
+	// The rollout strategy only watches the addon configurations defined in ClusterManagementAddOn.
+	// +kubebuilder:default={type: UpdateAll}
+	// +optional
+	RolloutStrategy RolloutStrategy `json:"rolloutStrategy,omitempty"`
+}
+
+// RolloutStrategy represents the rollout strategy of the add-on configuration.
+type RolloutStrategy struct {
+	// Type is the type of the rollout strategy, it supports UpdateAll, RollingUpdate and RollingUpdateWithCanary:
+	// - UpdateAll: when configs change, apply the new configs to all the selected clusters at once.
+	//   This is the default strategy.
```

Comment:
Just to confirm (no need to update the doc), this is the default for backwards compatibility, correct?  If so, would we consider changing the default to RollingUpdate in beta?

Reply:
Yes, it's for backwards compatibility. Good suggestion, we could consider using RollingUpdate as default in beta.